### ISSUE TITLE
VimwikiTOC and markdown

### ISSUE THREAD

**Issue by radio-van** [vimwiki/vimwiki#968]
My goal is to make nice-looking wiki in Markdown syntax, wich is both handy useful via Vim and via GitHub web-interface (thats why I've chosen markdown).
`.vimrc` settings:
```vim
let g:vimwiki_list = [{'path': '~/wiki/test/', 'syntax': 'markdown', 'ext': '.md', 'auto_toc': 0, 'auto_tags': 1}]
```
## issue
Auto-generated TOC (with `:VimwikiTOC` command) doesn't follow markdown syntax and therefore GitHub can't handle most of the links in TOC.

## reproducing
```markdown
link_to_external_file

# one
## two
## two
### three with spaces
```
According to markdown syntax, appropriate TOC with links that are correct on GitHub should be this:
```markdown
[link_to_external_file](path/to/external.md)
[one](#one) 
[two](#two)
[two](#1-two)
[three with spaces](#three-with-spaces)
```
but I get:
```markdown
[link_to_external_file](path/to/external)  # no .md ext, GitHub responses with 404 on such link
[one](#one)  # correct
[two](#one#two)  # not correct, GitHub won't follow such anchor
[two](#one#two)  # no counter for the header with the same text
[three with spaces](#three with spaces)  # no hyphens
```
Maybe I miss some option? I've tried both `master` and `dev` branches.
output of `:VimwikiShowVersion`:
```
Os: Linux
Vim: 802
Branch: dev
Revision: 1b16720
Date: 2020-08-02 03:57:22 -0400
```

**Comment by tinmarino** [comment]
> Maybe I miss some option?

Not really, just some months for theses issues (tracked here #664 were I add yours) to be solved.
Just 1 is already solved:
```markdown
[link_to_external_file](path/to/external)  # no .md ext, GitHub responses with 404 on such link
```
add `let g:vimwiki_markdown_link_ext = 1` __before__ `packadd vimwiki`

**Comment by tinmarino** [comment]
Fixed in dev: (428c60a: or Search multiple heading with same name)

Thanks for your issue recording us that there are users around.

Related: #666
